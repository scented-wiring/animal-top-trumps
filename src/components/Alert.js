import "../styles/Alert.css";

const Alert = ({ message, alertType, onAlertPress }) => {
  if (!message) return null;

  return (
    <div className="Alert">
      <h2>{alertType}!</h2>
      <p>{message}</p>
      <button className="clearAlert" onClick={onAlertPress}>
        OK
      </button>
    </div>
  );
};

export default Alert;
