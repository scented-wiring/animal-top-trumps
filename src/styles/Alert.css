.Alert {
  display: flex;
  flex-direction: column;
  margin: 200px auto 20px auto;
  padding-bottom: 20px;
  left: 0;
  right: 0;
  text-align: center;
  position: absolute;
  background-color: #385bac;
  width: 400px;
  color: white;
  border: 2px white solid;
  box-shadow: 10px 10px #004742;
}

.Alert p {
  margin-bottom: 23px;
}

.error-buttons button {
  margin-right: 10px;
  margin-left: 10px;
}

@media only screen and (max-width: 500px) {
  .Alert {
    width: 200px;
    margin-top: 100px;
  }
}
