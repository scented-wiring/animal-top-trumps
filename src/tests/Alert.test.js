import { render, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Alert from "../components/Alert";

describe("Alert component", () => {
  test("renders correctly", () => {
    const { asFragment } = render(<Alert />);
    expect(asFragment).toMatchSnapshot();
  });

  test("does not render an alert box if message prop is empty", () => {
    const { queryByText } = render(<Alert alertType="Success" />);
    expect(queryByText("Success!")).not.toBeInTheDocument();
  });

  test("renders an error message and back button", () => {
    const { getByText } = render(
      <BrowserRouter>
        <Alert alertType="Error" message="test" />
      </BrowserRouter>
    );
    expect(getByText("Error")).toBeInTheDocument();
    expect(getByText("test")).toBeInTheDocument();
    expect(getByText("BACK")).toBeInTheDocument();
  });

  test("renders a success message and ok button", () => {
    const { getByText } = render(
      <BrowserRouter>
        <Alert alertType="Success" message="test2" />
      </BrowserRouter>
    );
    expect(getByText("Success")).toBeInTheDocument();
    expect(getByText("test2")).toBeInTheDocument();
    expect(getByText("OK")).toBeInTheDocument();
  });

  test("renders a confirmation message and two buttons", () => {
    const { getByText } = render(
      <BrowserRouter>
        <Alert alertType="Are you sure?" message="test3" />
      </BrowserRouter>
    );
    expect(getByText("Are you sure?")).toBeInTheDocument();
    expect(getByText("test3")).toBeInTheDocument();
    expect(getByText("CANCEL")).toBeInTheDocument();
    expect(getByText("DELETE")).toBeInTheDocument();
  });

  test("alert clears on OK press", () => {
    const clearAlert = jest.fn();
    const { getByText } = render(
      <BrowserRouter>
        <Alert alertType="Success" message="test" setAlert={clearAlert} />
      </BrowserRouter>
    );
    fireEvent.click(getByText("OK"));
    expect(clearAlert).toHaveBeenCalled();
  });

  test("alert clears on BACK press", () => {
    const clearAlert = jest.fn();
    const { getByText } = render(
      <BrowserRouter>
        <Alert alertType="Are you sure?" message="test" setAlert={clearAlert} />
      </BrowserRouter>
    );
    fireEvent.click(getByText("CANCEL"));
    expect(clearAlert).toHaveBeenCalled();
  });

  test("alert clears on CANCEL press", () => {
    const clearAlert = jest.fn();
    const { getByText } = render(
      <BrowserRouter>
        <Alert alertType="Error" message="test" setAlert={clearAlert} />
      </BrowserRouter>
    );
    fireEvent.click(getByText("BACK"));
    expect(clearAlert).toHaveBeenCalled();
  });

  test("deletes all cards on confirmation press", () => {
    const deleteCards = jest.fn();
    const { getByText } = render(
      <BrowserRouter>
        <Alert
          alertType="Are you sure?"
          message="Delete all cards? This cannot be undone."
          deleteAllCards={deleteCards}
        />
      </BrowserRouter>
    );
    fireEvent.click(getByText("DELETE"));
    expect(deleteCards).toHaveBeenCalled();
  });
});
