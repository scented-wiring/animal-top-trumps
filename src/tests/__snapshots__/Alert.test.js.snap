// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component renders correctly 1`] = `[Function]`;
